api/db/populate-comments.js

```diff
@@ -0,0 +1,94 @@
+const txtgen = require('txtgen');
+
+const Event = require('../models/Event');
+const Attendant = require('../models/Attendant');
+const Comment = require('../models/Comment');
+
+let displayLogs;
+
+const prerequisites = (eventsCount, attendantsCount) => {
+  if (eventsCount === 0 && attendantsCount === 0) {
+    console.log(
+      '\x1b[33m',
+      'Attendants and events collection do not exist. Aborted comments collection population'
+    );
+    return false;
+  } else if (eventsCount === 0) {
+    console.log(
+      '\x1b[33m',
+      'Events collection does not exist. Aborted comments collection population'
+    );
+    return false;
+  } else if (attendantsCount === 0) {
+    console.log(
+      '\x1b[33m',
+      'Attendants collection does not exist. Aborted comments collection population'
+    );
+    return false;
+  }
+  return true;
+};
+
+const createComment = async (commentParams) => {
+  const comment = new Comment({
+    event: commentParams.event,
+    user: commentParams.user,
+    content: commentParams.content,
+    parentComment: commentParams.parentComment,
+  });
+  await comment.save();
+
+  if (displayLogs) {
+    console.log('\n', '\x1b[0m', `New comment created: ${comment}`);
+  }
+  return comment;
+};
+
+const createComments = async (verbose) => {
+  console.log('\n', '\x1b[0m', 'Populating comments collection...');
+  displayLogs = verbose;
+
+  const eventsList = await Event.distinct('_id');
+  const attendantsList = await Attendant.find();
+  const eventsCount = eventsList.length;
+  const attendantsCount = attendantsList.length;
+
+  if (!prerequisites(eventsCount, attendantsCount)) return;
+
+  const commentsArray = [];
+  for (const event of eventsList) {
+    // Find attendants for event
+    const eventAttendants = attendantsList.filter(
+      (attendant) => attendant.event.toString() === event.toString()
+    );
+    const eventComments = await Comment.find({ event });
+
+    // Number of comments to generate
+    const commentCount = Math.floor(Math.random() * 20);
+    for (let i = 0; i <= commentCount; i++) {
+      const user =
+        eventAttendants[Math.floor(Math.random() * eventAttendants.length)]
+          .user;
+      const content = txtgen.paragraph(Math.ceil(Math.random() * 3));
+      const parentComment =
```

Comment:
A few issues with this:
- Only top-level comments can be parent comments of other comments, so we should enforce this here somehow
- What happens when `Math.random() <= 0.2` is `true` but `eventComments` is still empty? We should probably skip this altogether if `eventComments` is empty
- Should we make it more probable for a comment to have a parent comment? I guess most real conversation would occur in comment threads, no?
- If a comment doesn't have a parent comment, the `parentComment` value should not be set instead of being set to `null`. This can be easily fixed by only including the value on the `commentParams` object if it's not `null`

Reply:
1. I don't think I follow. 
2. If the event comments is empty, then parent comment will be null, which means it will just be a 'root' comment. This is just to simulate the feel for having replies to comments. 
3. What would you like it set to?

Reply:
2. I see that in this case the `parentComment` is `undefined`, while in the other case is `null`. This is related to point 4, so I'll comment more below
3. Maybe `>= 0.5`? What do you think?
4. I think that the problem with this as is is that pushing a `null` value makes the document invalid, as it's not as defined on the comment schema and it wouldn't pass the validations that we have in place